pkg/storaged/anaconda.jsx

```diff
@@ -19,10 +19,105 @@
+    if (!client.in_anaconda_mode())
+        return;
+
+    try {
+        const passphrases = JSON.parse(window.sessionStorage.getItem("cockpit_passphrases")) || { };
+        passphrases[decode_filename(block.Device)] = passphrase;
+        window.sessionStorage.setItem("cockpit_passphrases", JSON.stringify(passphrases));
+    } catch {
+        console.warn("Can't record passphrases");
+    }
+}
+
+export function export_mount_point_mapping() {
+    if (!client.in_anaconda_mode())
+        return;
+
+    let passphrases = { };
+    try {
+        passphrases = JSON.parse(window.sessionStorage.getItem("cockpit_passphrases")) || { };
+    } catch {
+        console.warn("Can't read passphrases");
+    }
+
+    function fstab_info(config) {
+        let dir;
+        let subvols;
+
+        for (const c of config) {
+            if (c[0] == "fstab") {
+                const d = client.strip_mount_point_prefix(decode_filename(c[1].dir.v));
+                if (d) {
+                    const sv = parse_subvol_from_options(decode_filename(c[1].opts.v));
+                    if (sv) {
+                        if (sv.pathname) {
+                            if (!subvols)
+                                subvols = { };
+                            subvols[sv.pathname] = d;
+                        }
+                    } else if (!dir) {
+                        dir = d;
+                    }
+                }
+            }
+        }
+
+        if (dir || subvols)
+            return {
+                type: "filesystem",
+                dir,
+                subvolumes: subvols
+            };
+    }
+
+    function block_info(block) {
+        if (block.IdUsage == "filesystem") {
+            return fstab_info(block.Configuration);
+        } else if (block.IdUsage == "swap") {
+            return {
+                type: "swap",
+            };
+        } else if (block.IdUsage == "crypto") {
+            const cleartext_block = client.blocks_cleartext[block.path];
+
+            let content_info;
+            if (cleartext_block)
+                content_info = block_info(cleartext_block);
+            else {
+                const block_crypto = client.blocks_crypto[block.path];
+                if (block_crypto)
+                    content_info = fstab_info(block_crypto.ChildConfiguration);
+            }
+
+            if (content_info)
+                return {
+                    type: "crypto",
+                    cleartext_device: cleartext_block && decode_filename(cleartext_block.Device),
+                    content: content_info,
+                    passphrase: passphrases[decode_filename(block.Device)],
+                };
+        }
+    }
+
+    const mpm = { };
+    for (const p in client.blocks) {
+        const b = client.blocks[p];
+        mpm[decode_filename(b.Device)] = block_info(b);
+    }
+
+    window.sessionStorage.setItem("cockpit_mount_points", JSON.stringify(mpm));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19924-20240202-090532-99f6d966-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)